### ISSUE TITLE
flash attention triton Segmentation fault

### ISSUE THREAD

**Issue by zhangbin1997** [Dao-AILab/flash-attention#740]
Hello, I encountered an Segmentation fault error while executing the second line of code below，

  from flash_attn.flash_attn_triton import flash_attn_func
  output = flash_attn_func(
      q,
      k,
      v,
      None,
      is_causal,
      self.softmax_scale,
  )

Here are the error details：

Caught signal 11 (Segmentation fault: address not mapped to object at address 0x7ffc90000000)
==== backtrace (tid:1479935) ====
 0 0x0000000000042520 __sigaction()  ???:0
 1 0x00000000000a54c9 free()  ???:0
 2 0x00000000006731c5 std::__detail::_Compiler<std::regex_traits<char> >::_Compiler()  ???:0
 3 0x000000000065cdfc triton::driver::path_to_ptxas()  ???:0
 4 0x00000000006b24fa pybind11::cpp_function::initialize<init_triton_codegen(pybind11::module_&&)::{lambda(backend_t, triton::ir::module&, unsigned long, int, int, pybind11::dict&, unsigned long)#1}, std::tuple<std::string, std::map<std::string, pybind11::object, std::less<std::string>, std::allocator<std::pair<std::string const, pybind11::object> > >, int>, backend_t, triton::ir::module&, unsigned long, int, int, pybind11::dict&, unsigned long, pybind11::name, pybind11::scope, pybind11::sibling, pybind11::return_value_policy>(init_triton_codegen(pybind11::module_&&)::{lambda(backend_t, triton::ir::module&, unsigned long, int, int, pybind11::dict&, unsigned long)#1}&&, std::tuple<std::string, std::map<std::string, pybind11::object, std::less<std::string>, std::allocator<std::pair<std::string const, pybind11::object> > >, int> (*)(backend_t, triton::ir::module&, unsigned long, int, int, pybind11::dict&, unsigned long), pybind11::name const&, pybind11::scope const&, pybind11::sibling const&, pybind11::return_value_policy const&)::{lambda(pybind11::detail::function_call&)#3}::_FUN()  triton.cc:0
 5 0x00000000006a0661 pybind11::cpp_function::dispatcher()  :0
 6 0x000000000015fe0e PyObject_CallFunctionObjArgs()  ???:0
 7 0x00000000001565eb _PyObject_MakeTpCall()  ???:0
 8 0x000000000014fe0a _PyEval_EvalFrameDefault()  ???:0
 9 0x000000000016070c _PyFunction_Vectorcall()  ???:0
10 0x0000000000208dd2 _PyLong_Format()  ???:0
11 0x000000000014f766 _PyEval_EvalFrameDefault()  ???:0
12 0x000000000016070c _PyFunction_Vectorcall()  ???:0
13 0x0000000000208dd2 _PyLong_Format()  ???:0
14 0x000000000014fd66 _PyEval_EvalFrameDefault()  ???:0
15 0x000000000016070c _PyFunction_Vectorcall()  ???:0
16 0x000000000016f192 PyObject_Call()  ???:0
17 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
18 0x000000000016e4e1 PyMethod_New()  ???:0
19 0x000000000016f192 PyObject_Call()  ???:0
20 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
21 0x000000000016070c _PyFunction_Vectorcall()  ???:0
22 0x000000000016f192 PyObject_Call()  ???:0
23 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
24 0x000000000016070c _PyFunction_Vectorcall()  ???:0
25 0x0000000000208dd2 _PyLong_Format()  ???:0
26 0x000000000014fd66 _PyEval_EvalFrameDefault()  ???:0
27 0x000000000016070c _PyFunction_Vectorcall()  ???:0
28 0x000000000074a64e THPFunction_apply()  ???:0
29 0x000000000015fe38 PyObject_CallFunctionObjArgs()  ???:0
30 0x000000000016f12b PyObject_Call()  ???:0
31 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
32 0x000000000016e4e1 PyMethod_New()  ???:0
33 0x0000000000208dd2 _PyLong_Format()  ???:0
34 0x000000000014f766 _PyEval_EvalFrameDefault()  ???:0
35 0x000000000016e4e1 PyMethod_New()  ???:0
36 0x000000000016f192 PyObject_Call()  ???:0
37 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
38 0x000000000016e4e1 PyMethod_New()  ???:0
39 0x000000000016f192 PyObject_Call()  ???:0
40 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
41 0x000000000016070c _PyFunction_Vectorcall()  ???:0
42 0x000000000015582d _PyObject_FastCallDictTstate()  ???:0
43 0x000000000016b54c _PyObject_Call_Prepend()  ???:0
44 0x00000000002841e0 PyInit__datetime()  ???:0
45 0x00000000001565eb _PyObject_MakeTpCall()  ???:0
46 0x000000000014fd66 _PyEval_EvalFrameDefault()  ???:0
47 0x000000000016e4e1 PyMethod_New()  ???:0
48 0x000000000016f192 PyObject_Call()  ???:0
49 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
50 0x000000000016e4e1 PyMethod_New()  ???:0
51 0x000000000016f192 PyObject_Call()  ???:0
52 0x000000000014b2c1 _PyEval_EvalFrameDefault()  ???:0
53 0x000000000016070c _PyFunction_Vectorcall()  ???:0
54 0x000000000015582d _PyObject_FastCallDictTstate()  ???:0
55 0x000000000016b54c _PyObject_Call_Prepend()  ???:0
56 0x00000000002841e0 PyInit__datetime()  ???:0
=================================
[2023-12-27 13:34:11,118] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: -11) local_rank: 0 (pid: 1479935) of binary: /bin/python

May I ask what the problem is?
My environment is as follows：
triton                        2.0.0.dev20221202
torch                         2.1.0a0+32f93b1
cuda-python                   12.2.0rc5+5.g84845d1

**Comment by tridao** [comment]
Yeah idk what's wrong I haven't been using the triton version recently.
